knime_extension/src/nodes/transform.py

```diff
@@ -623,3 +623,168 @@ def execute(self, exec_context: knext.ExecutionContext, input_1):
             size=gdf[self.num_col], method="uniform", seed=seed
         )
         return knut.to_table(gdf, exec_context)
+
+
+############################################
+# Directed Bezier Curve
+############################################
+
+
+@knext.node(
+    name="Directed Bezier Curve",
+    node_type=knext.NodeType.MANIPULATOR,
+    icon_path="icons/icon/GeometryTransformation/ODtoCurve.png",
+    category=category,
+    after="",
+)
+@knext.input_table(
+    name="Geo table",
+    description="Input table containing geometry columns representing origin and destination points.",
+)
+@knext.output_table(
+    name="Table with directed Bézier curve",
+    description="Output table with geometry columns representing Bézier curves connecting origin and destination points.",
+)
+@knut.geo_node_description(
+    short_description="Generate Bézier curves between origin and destination points.",
+    description="""This node generates a GeoSeries containing geometries representing smooth Bézier curves between origin and destination points. 
+    The Bézier curves are created based on user-defined parameters,including the number of points, height scaling, and curve angle.
+    This transformation is particularly useful for visualizing flows or movements in a more intuitive manner compared to straight lines.    
+    """,
+    references={
+        "Bézier curve": "https://en.wikipedia.org/wiki/B%C3%A9zier_curve",
+    },
+)
+class ODtoCurveNode:
+    """
+    This node generates geometries representing connections between origin and destination points as Bézier curves.
+    """
+
+    o_geo_col = knext.ColumnParameter(
+        "Origin geopoint column",
+        "Select the geometry column that describes the origins.",
+        port_index=0,
+        column_filter=knut.is_geo_point,
+        include_row_key=False,
+        include_none_column=False,
+    )
+
+    d_geo_col = knext.ColumnParameter(
+        "Destination geopoint column",
+        "Select the geometry column that describes the destination.",
+        port_index=0,
+        column_filter=knut.is_geo_point,
+        include_row_key=False,
+        include_none_column=False,
+    )
+    result_settings = knut.ResultSettings(
```

Comment:
Why do you use the ResultSettings class here? This class was intended for single input column geometry nodes.

Reply:
revise to appending column